### ISSUE TITLE
Q: How to configure/remove Abseil?

### ISSUE THREAD

**Issue by JCash** [protocolbuffers/protobuf#10431]
I'm trying to build protobuf for a very popular platform (under NDA), and as such, one cannot always rely on where certain headers are, if they're complete or what defines are used by the ocmpiler.

E.g. the define FreeBSD happens to be set, but the header isn't available:
```
#if defined(__APPLE__) || defined(__FreeBSD__)
#include <sys/sysctl.h>
#endif
```
(and the list goes on ofc...)

In particular, I cannot opt out of the Abseil situation, and it doesn't seem to be configurable. at all?

How do you propose I make this work with Abseil?
Is there an option to use protobuf now without it?

Or am I forced to use an older version of protobuf?
v21.5 seems to be without Abseil, but it is also stuck at C++ 11, and my compiler requires C++14+.

Please advise.

**Comment by fowles** [comment]
The dependency on Abseil is not conditional and the 22.x release line will have it.  This is a very important change for removing literal decades of tech debt from protobuf.

In the short term, you can say on the 21.x release line which does not have an abseil dep (and won't).  In the medium term, you can contribute patches to abseil and protobuf to make them work on your system.

Both protobuf and abseil publish their supported platform criteria at https://opensource.google/documentation/policies/cplusplus-support.  If your platform is not officially supported it would fall under "best effort" and would likely require contributions to ensure continued compatibility.

**Comment by JCash** [comment]
Thanks for the info!
Yeah, I was thinking of doing contributions, but wasn't sure of if they're welcome or not.
For the time being, I dodged the immedate issue by disabling one of our unit tests, and simply modifying some headers.

I had to make three changes, one is fairly straightfroward:
E.g I detect the define `GOOGLE_USE_PROTOBUF_LITTLE_ENDIAN`, which then sets `PROTOBUF_LITTLE_ENDIAN`.
This way I can sidestep the entire "endian.h" bonanza.

I'll make some PR's when I'm back from Gamescom in a week or so.
